Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -290,12 +289,10 @@ IndexInfo::globalSpectrumIndicesFromDetectorIndices(
         const auto detectorIndex = static_cast<size_t>(spectrumDefinition);
         if (detectorMap.size() > detectorIndex &&
             detectorMap[detectorIndex] != 0) {
-          if (detectorMap[detectorIndex] > 1)
-            throw std::runtime_error(
-                "Multiple spectra correspond to the same detector");
-          // Increment flag to catch two spectra mapping to same detector.
-          ++detectorMap[detectorIndex];
           spectrumIndices.push_back(i);
+          if (detectorMap[detectorIndex] == 1) {
+            ++detectorMap[detectorIndex];
+          }
```

Comment:
I am not very sure that I understand the MPI logic here, but my understanding of the original code was the following: `detectorMap[detectorIndex]` is incremented only once, when it is 1, so it becomes 2, and the next time it throws. Is this not the case? Is `detectorMap[detectorIndex]` used in an other way than checking if it grows over 1? Otherwise, this preserves the original logic, it is incremented from 1 to 2, and not any further, but there is no throw at this point. (see next comment)

Reply:
I think you are correct, I misunderstood the way `detectorMap` was initialized.